rivertest/worker.go

```diff
@@ -0,0 +1,166 @@
+	t.Helper()
+	return w.WorkOpts(ctx, t, args, nil)
+}
+
+func (w *Worker[T, TTx]) WorkJob(ctx context.Context, t *testing.T, job *river.Job[T]) error {
+	t.Helper()
+	return w.WorkJobOpts(ctx, t, job, nil)
+}
+
+func (w *Worker[T, TTx]) WorkJobOpts(ctx context.Context, t *testing.T, job *river.Job[T], opts *river.InsertOpts) error {
+	t.Helper()
+
+	// populate river client into context:
+	ctx = WorkContext(ctx, w.client)
+
+	doInner := w.buildMiddlewareChain(job)
+	return doInner(ctx)
+}
+
+func (w *Worker[T, TTx]) WorkOpts(ctx context.Context, t *testing.T, args T, opts *river.InsertOpts) error {
+	t.Helper()
+
+	job, err := makeJobFromArgs(t, w.config, args, opts)
+	if err != nil {
+		return err
+	}
+
+	return w.WorkJobOpts(ctx, t, job, opts)
+}
+
+func (w *Worker[T, TTx]) buildMiddlewareChain(job *river.Job[T]) func(ctx context.Context) error {
+	// apply global middleware from client, as well as worker-specific middleware
+	workerMiddleware := w.worker.Middleware(job)
+
+	doInner := func(ctx context.Context) error {
+		jobTimeout := w.worker.Timeout(job)
+		if jobTimeout == 0 {
+			jobTimeout = w.config.JobTimeout
+		}
+
+		// No timeout if a -1 was specified.
+		// TODO: is this timeout being applied in the wrong place? Should it apply
+		// on the *outside* of all middleware?
+		if jobTimeout > 0 {
+			var cancel context.CancelFunc
+			ctx, cancel = context.WithTimeout(ctx, jobTimeout)
+			defer cancel()
+		}
+
+		return w.worker.Work(ctx, job)
+	}
+
+	allMiddleware := make([]rivertype.WorkerMiddleware, 0, len(w.config.WorkerMiddleware)+len(workerMiddleware))
+	allMiddleware = append(allMiddleware, w.config.WorkerMiddleware...)
+	allMiddleware = append(allMiddleware, workerMiddleware...)
+
+	if len(allMiddleware) > 0 {
+		// Wrap middlewares in reverse order so the one defined first is wrapped
+		// as the outermost function and is first to receive the operation.
+		for i := len(allMiddleware) - 1; i >= 0; i-- {
+			middlewareItem := allMiddleware[i] // capture the current middleware item
+			previousDoInner := doInner         // Capture the current doInner function
+			doInner = func(ctx context.Context) error {
+				return middlewareItem.Work(ctx, job.JobRow, previousDoInner)
+			}
+		}
+	}
+
+	return doInner
+}
+
+var idSeq int64 = 0 //nolint:gochecknoglobals
+
+func nextID() int64 {
+	return atomic.AddInt64(&idSeq, 1)
+}
+
+func makeJobFromArgs[T river.JobArgs](t *testing.T, config *river.Config, args T, opts *river.InsertOpts) (*river.Job[T], error) {
+	t.Helper()
+
+	if opts == nil {
+		opts = &river.InsertOpts{}
+	}
+
+	created := time.Now().Add(-1500 * time.Millisecond)
+
+	encodedArgs, err := json.Marshal(args)
+	require.NoError(t, err)
+
+	// Round trip the args to validate JSON marshalling/unmarshalling on the type.
+	var decodedArgs T
+	if err = json.Unmarshal(encodedArgs, &decodedArgs); err != nil {
+		return nil, err
+	}
+
+	// TODO: use insertParamsFromConfigArgsAndOptions instead of reimplementing it here?
```

Comment:
@brandur looked at this one a bit, and `insertParamsFromConfigArgsAndOptions` can't really be extracted from the main package because there are a number of dependencies on its own types, including interfaces whose methods return structs in that package. There's no way to avoid circular dependencies.

Options I see:

1. Live with the ugliness of exporting this API even though it's not useful or intended for public usage
2. Essentially reimplement its logic here to create realistic test jobs
3. Instead use `rivershared/testfactory` for building test jobs (wrap `testfactory.Job_Build` to return a `*Job[T]` instead of `*riverdriver.JobInsertFullParams`.